sign/options.go

```diff
@@ -49,7 +58,25 @@ type Options struct {
 // Default returns a default Options instance.
 func Default() *Options {
 	return &Options{
-		Timeout:  3 * time.Minute,
-		PassFunc: generate.GetPass,
+		Timeout:              3 * time.Minute,
+		PassFunc:             generate.GetPass,
+		EnableTokenProviders: true,
+	}
+}
+
+// VerifySignOptions checks that options have the minimum settings
+// for signing files or images:
+func (o *Options) VerifySignOptions() error {
+	// Our library is not designed to run in interactive mode
+	// this means that we will only support signing if we get a keypair or
+	// identity token to run keyless signing:
+	if o.PrivateKeyPath != "" && o.IdentityToken == "" && !o.EnableTokenProviders {
+		return errors.New("signing can only be done if a key or identity token are set")
+	}
+
+	// Ensure that the private key file exists
+	if o.PrivateKeyPath != "" && !util.Exists(o.PrivateKeyPath) {
```

Comment:
Add `util.Exists` to the `impl interface` to be unit-testable.

Reply:
Done, added the unit test too